pkg/networkservice/common/connect/server_test.go

```diff
@@ -264,6 +310,66 @@ func TestConnectServer_RequestParallel(t *testing.T) {
 		require.Equal(t, int32(parallelCount), serverClient.count)
 		require.Equal(t, int32(parallelCount), serverA.count)
 		require.Equal(t, int32(parallelCount), serverNext.count)
+
+		goleak.VerifyNone(t, ignoreCurrentGoroutines)
+	}()
+}
+
+func TestConnectServer_RequestFail(t *testing.T) {
+	defer goleak.VerifyNone(t, goleak.IgnoreCurrent())
+
+	// 1. Create connectServer
+
+	s := connect.NewServer(context.TODO(),
+		func(_ context.Context, cc grpc.ClientConnInterface) networkservice.NetworkServiceClient {
+			return injecterror.NewClient()
+		},
+		grpc.WithInsecure(),
+	)
+
+	func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		// 2. Setup servers
+
+		urlA := &url.URL{Scheme: "tcp", Host: "127.0.0.1:10000"}
+		serverA := new(captureServer)
+
+		err := startServer(ctx, urlA, next.NewNetworkServiceServer(
+			serverA,
+			newEditServer("a", "A", &networkservice.Mechanism{
+				Cls:  cls.LOCAL,
+				Type: kernel.MECHANISM,
+			}),
+		))
+		require.NoError(t, err)
+
+		// Time for servers to start listening
+		<-time.After(10 * time.Millisecond)
```

Comment:
Shouldn't this be replaced with `checkServer`?

Reply:
Done